src/document/DocumentAnnotator.ts

```diff
@@ -97,20 +100,23 @@ export default class DocumentAnnotator extends BaseAnnotator {
         }
     };
 
-    handleSelectionChange = (): void => {
-        if (this.selectionChangeTimer) {
-            clearTimeout(this.selectionChangeTimer);
-        }
-
-        // Clear current selection immediately
-        this.store.dispatch(setSelectionAction(null));
-
-        // Wait Pdf.js textLayer enhancement for 300ms (they hardcode this magic number)
-        this.selectionChangeTimer = setTimeout(() => {
-            this.store.dispatch(setSelectionAction(getSelectionItem()));
-            this.selectionChangeTimer = null;
-        }, 300);
-    };
+    handleSelectionChange = debounce(
+        (): void => {
+            // Clear current selection immediately
+            this.store.dispatch(setSelectionAction(null));
+
+            // Wait Pdf.js textLayer enhancement for 300ms (they hardcode this magic number)
+            this.selectionChangeTimer = window.setTimeout(() => {
+                this.store.dispatch(setSelectionAction(getSelectionItem()));
+                this.selectionChangeTimer = null;
+            }, TEXT_LAYER_ENHANCEMENT);
+        },
+        TEXT_LAYER_ENHANCEMENT,
```

Comment:
I set the option `leading: true`, which means it fires at 0ms and will prevent next call in 300ms

Reply:
Should trailing also be `true`? Would this selectionchange event also be used when explicitly in highlight mode? I'm wondering about the responsiveness of the highlighting if it needs to wait 300ms constantly

Reply:
If trailing is `true` and the second selection is made within in 300ms, the function will be called three times: start of the first selection, end of the first selection, and the start of the second selection. The end of the first selection and the start of the second selection trigger at the same time. So the trailing is not necessary.

Yeah I guess we'll use it when explicitly in highlight mode too. I think we can adjust this parameter when we implement the highlighting in highlight mode because at that time we can try and feel the responsiveness.

Reply:
I think the logic in here is a little confusing due to the combination of debounce with setTimeout and the fact that they use the same duration. Should we separate and debounce the reset/set logic independently?